### DISCUSSION TITLE
프로토타입 목표 설정, 프로토타입 구현을 위한 사전 질문

### DISCUSSION THREAD

**Discussion by gyutae100** [InnerCircleA/design-system#4] [General]
# 프로토타입 목표 설정, 프로토타입 구현을 위한 사전 질문

## 일정 상, 22-12-15(목)~22-12-21(수)까지 팀 과제로 다음을 수행하게 됩니다.

1. 프로토타입 목표 설정
2. 프로토타입 구현
 
 ## 이를 위한 사전 질문을 드립니다.
 - Q1. 이전에 프로토타입 산출물을 위한 과정을 타 직군과 함께 진행한 경험은? 
 - Q2. 프로토타입 도구를 통해서 직접 디자인 산출물을 제작 해본 경험은? (예시로, Figma 활용)
 - Q3. 프로토타입을 활용한 최종 산출물인, 과제의 최종 목적인 프로덕트는 무엇을 만들고 싶으신지?  (예시로, 배민 사장님 서비스의 백오피스)
   
 ## 과제 Key Point
 1. 적합한 프로토타입 목표 선정하기 
 2. 프로토타입 목표에 적합한 프로토타입 과정 선정하기 
 3. 프로토타입 과정 중 산출할 프로토타입을 만드는 도구 선정하기
 
 ## 이상입니다.

**Comment by peanudge** [comment]
> Q1. 이전에 프로토타입 산출물을 위한 과정을 타 직군과 함께 진행한 경험은?
> Q2. 프로토타입 도구를 통해서 직접 디자인 산출물을 제작 해본 경험은?

저는 질문 두개에 같이 답변할게요.

이전에 디자이너랑 토이프로젝트를 해본 경험이 있었는데 대부분 디자이너가 이끌어서 피그마에 작은 범위의 디자인 시스템을 만들고 그 위에서 개발을 시작한 경험은 있어요. (아래와같이)

<img width="300" alt="스크린샷 2022-12-07 오후 10 56 52" src="https://user-images.githubusercontent.com/20085849/206198254-859a2189-b93d-4f38-aea7-0abd7f8a664c.png">

<img width="400" alt="스크린샷 2022-12-07 오후 10 56 45" src="https://user-images.githubusercontent.com/20085849/206198373-a5103e6f-9c95-4b1f-96a4-bbf8080491e9.png">

근데 저희가 피그마 위에서 하는 **그런 바탕 작업에 시간을 많이 투자해야할까**에 대한 고민은 있네요. 물론 도움은 되지만 그런 작업 자체가 **프론트엔드 개발자가 디자인 시스템에 대한 이해**를 하는데 큰 의미가 있을까 싶기도해요. 저는 스토리북에서 위에서 코드 레벨에서 바로 같이 시작해봐도 좋지않을까 생각이 드네요.

> Q3. 프로토타입을 활용한 최종 산출물인, 과제의 최종 목적인 프로덕트는 무엇을 만들고 싶으신지? (예시로, 배민 사장님 서비스의 백오피스)

프로토타입을 아주 조금만들고 바로 개발하고 이런 짧게 경험을 현재 하고 있긴한데 질문을 잘 이해를 못하겠어요. **바닥부터해서 프로젝트를 완성해본 경험을 물어보시는걸까요?** 그렇다면 No 입니다. 🤦‍♂️

**Reply by gyutae100** [reply]
## 프로토타입 없이, 바로 스토리북에 개발하자는 제안에 대해서
- 디자이너 관점에서, 프로토타입 산출물은 크게 Low-fi 프로토타입과 High-fi 프로토타입으로 구분됩니다.
- `Low-fi 프로토타입`은 낮은 구체화 수준으로, `종이`부터 존재하고,
- `High-fi 프로토타입`은 높은 구체화 수준으로, `피그마`가 좋은 예 같습니다.
- 참고로 중간 레벨의 프로토타입은 `와이어 프레임`이 있습니다.
- **제 개인적인 추측으로는, 이번 교육 과정이 개발자가 디자인 시스템을 구축하기 위해서는, 목표인 제품의 시작 단계인 프로토타입 부터 경험해보는 것을 권장하는 것으로 추측됩니다.**
- **프로토타입을 기반으로, 파운데이션 디자인 시스템과 컴포넌트 디자인 시스템 중 하나를 채택 가능하다고 보입니다.**
- 근거는 다음 과제의 산출물은, 프로토타입 목표 설정과 프로토타입 구현이 있습니다.
- 만약 스토리북 단계에서 바로 구현을 한다면, 이는 프로토타입 없이 바로 구현함을 의미합니다.
- 프로토타입 없이 개발한다면, 그 과정 속에서 프로토타입의 중요성을 파악하는 계기가 되지 않을까 조심스럽게 추측됩니다.
- 물론 저희처럼 백오피스를 최종 산출물로 가정한 경우에는, 프로토타입이 없이 구현하는 방법이 `적은 비용`으로 `생산성`을 확보하는 방식일 수 있겠다는 판단이 되기도 합니다.
- 그 경우에는 피그마 시안을 제공하는 오픈소스 중 컴포넌트 기반의 디자인 시스템을 참고해서, 해당 디자인 시스템을 구성하는 각각의 컴포넌트의 형태와 기능, 컴포넌트 분류 기준, 컴포넌트 간 합성 또는 협력 방법, 컴포넌트의 역할과 책임 등을 어느 범위 까지를 우리 과제에 포함시키는 부분을 고민하면 될 것으로 보입니다.
- **결론으로 사전에 우선적으로, 백오피스 중 어느 서비스의 백오피스 구현 할 것인지를 먼저 정하고, 해당 서비스의 백오피스에 필요한 최소한의 비용을 주목적으로 작업 범위를 구분하면 좋을 것 같습니다.**
- **개인적인 견해로, 백오피스가 아닌, 사용자 서비스 제작을 프로덕트를 목적으로 프로토타입 과정을 진행하는 것 또한, 디자인 시스템을 이해하는 좋은 과정이라고 생각합니다.**

## 그 외
1. 답변 감사합니다.
2. Q3은 제가 의미를 모호하게 작성해서, 현재는 문구를 수정 완료했습니다.
